This notebook shows how we can find the images that contain labels with high frequency. Note that I only look at all the human labels because they have high confidence, and in total, they are much less than machine labels. 

In [359]:
import pandas as pd
import numpy as np


In [360]:
# Read all the human label

train_human_labels = pd.read_csv('/media/yshi/4TB/openImage/metadata/train_human_labels.csv', dtype = {'ImageID': str, "Source": str, 'LabelName':str, "Confidence": np.float64})

In [361]:
train_human_labels.shape

(8036466, 4)

So there are 8036466 labels in total. (Different from the total number of images because some images might contain more than one labels.) Next, we load the total trainable classes.

In [362]:
label_descriptions = pd.read_csv('/media/yshi/4TB/openImage/metadata/classes-trainable.csv')

In [363]:
label_descriptions.shape

(7178, 1)

In [365]:
# trainable_labels = list(label_descriptions.label_code)

Next, we load class descriptions, in order to make sure later in the results, all labels with high frequencies are sensible categories.

In [369]:
class_descriptions = pd.read_csv('/media/yshi/4TB/openImage/metadata/class-descriptions.csv', dtype=str)

In [370]:
lable_code_to_description = dict(zip(class_descriptions.label_code, class_descriptions.description))

In [371]:
class_descriptions.head()

label_code          description
0  /m/0100nhbf     Sprenger's tulip
1  /m/0104x9kv             Vinegret
2  /m/0105jzwx            Dabu-dabu
3  /m/0105ld7g  Pistachio ice cream
4  /m/0105lxy5                 Woku

Some descriptions corresponds with more than 1 unique label codes. That might be something we should consider in the future. 

In [375]:
class_description_counts = class_descriptions.groupby('description').count().rename(columns={'label_code':'count'})

In [376]:
repeated_descriptions = class_description_counts[(class_description_counts > 1).values].sort_values('count', ascending=False)

In [377]:
repeated_descriptions

count
description                
Trunk                     4
Wedge                     3
Referee                   3
Crop                      3
Punch                     3
Morning glory             3
Lilac                     3
Eel                       3
Mint                      3
Windflower                3
Liver                     3
Squash                    3
Alfa romeo giulietta      3
Snapper                   3
Polar bear                2
Powder                    2
Raven                     2
Prickly pear              2
Propeller                 2
Quail                     2
Rainbow                   2
Ratchet                   2
Poison ivy                2
Rattlesnake               2
Adapter                   2
Red salamander            2
Ringlet                   2
Robin                     2
Poinsettia                2
Rolls-royce corniche      2
...                     ...
Fiat                      2
Fiat 500                  2
Filter                    2
Fire-bellied toad         2
Fish                      2
Flash                     2
Flute                     2
Foam                      2
Foil                      2
Ford                      2
Ford escort               2
Ford fairmont             2
Ford fusion               2
Ford granada              2
Ford landau               2
Ford model a              2
Freeride                  2
Friendship day            2
Geranium                  2
Gimlet                    2
Glider                    2
Gold                      2
Goldenrod                 2
Grasshopper               2
Greyhound                 2
Groom                     2
Guarana                   2
Harrier                   2
Harvester                 2
Zombie                    2

[268 rows x 1 columns]

https://stackoverflow.com/questions/17071871/select-rows-from-a-dataframe-based-on-values-in-a-column-in-pandas

In [381]:
# find all labels that have are in the trainable dataframe
trainable_human_all = train_human_labels.loc[train_human_labels['LabelName'].isin(label_descriptions['label_code'])]

In [311]:
# select all the training images that start with 0, 1, 2, or 3 (because they are all the data I've downloaded so far)
trainable_human_all = trainable_human_all[trainable_human_all['ImageID'].str.match('[0-3]')]

In [312]:
trainable_human_all

ImageID                    Source   LabelName  Confidence
0        000002b66c9c498e  crowdsource-verification   /m/01kcnl         1.0
1        000002b66c9c498e              verification    /m/012mj         1.0
2        000002b66c9c498e              verification   /m/012yh1         1.0
3        000002b66c9c498e              verification   /m/014sv8         1.0
4        000002b66c9c498e              verification   /m/016c68         1.0
5        000002b66c9c498e              verification   /m/016q19         1.0
6        000002b66c9c498e              verification   /m/019nj4         1.0
7        000002b66c9c498e              verification   /m/019_nn         1.0
8        000002b66c9c498e              verification   /m/019sc6         1.0
10       000002b66c9c498e              verification   /m/01bsxb         1.0
12       000002b66c9c498e              verification   /m/01d40f         1.0
13       000002b66c9c498e              verification   /m/01d74z         1.0
14       000002b66c9c498e              verification   /m/01fklc         1.0
15       000002b66c9c498e              verification   /m/01g317         1.0
16       000002b66c9c498e              verification   /m/01mzpv         1.0
17       000002b66c9c498e              verification    /m/01tv9         1.0
18       000002b66c9c498e              verification    /m/01vkl         1.0
19       000002b66c9c498e              verification    /m/01ykh         1.0
20       000002b66c9c498e              verification   /m/021dfr         1.0
21       000002b66c9c498e              verification  /m/026pc7x         1.0
22       000002b66c9c498e              verification    /m/0270h         1.0
23       000002b66c9c498e              verification    /m/0271t         1.0
24       000002b66c9c498e              verification  /m/0283dt1         1.0
25       000002b66c9c498e              verification    /m/0284d         1.0
26       000002b66c9c498e              verification    /m/02cwm         1.0
27       000002b66c9c498e              verification   /m/02czv8         1.0
28       000002b66c9c498e              verification   /m/02f6b9         1.0
29       000002b66c9c498e              verification   /m/02jwqh         1.0
30       000002b66c9c498e              verification  /m/02p0tk3         1.0
31       000002b66c9c498e              verification   /m/02qm2x         1.0
...                   ...                       ...         ...         ...
2833141  3fffb6e78869527f              verification    /m/07k1x         1.0
2833142  3fffb6e78869527f              verification    /m/07yv9         1.0
2833143  3fffbb148106d5a9              verification   /m/01g317         1.0
2833144  3fffbb148106d5a9              verification   /m/01n5jq         1.0
2833145  3fffbb148106d5a9              verification   /m/0463sg         1.0
2833146  3fffbb148106d5a9              verification    /m/09j2d         1.0
2833147  3fffbb148106d5a9              verification    /m/0dzct         1.0
2833148  3fffbb87bc1242a0              verification   /m/01g317         1.0
2833149  3fffbb87bc1242a0              verification   /m/01gkx_         1.0
2833151  3fffbb87bc1242a0              verification    /m/05s2s         1.0
2833152  3fffbb87bc1242a0              verification    /m/07j7r         1.0
2833153  3fffbb87bc1242a0              verification    /m/09j2d         1.0
2833154  3fffc08eddf9167f              verification   /m/0138tl         1.0
2833155  3fffc08eddf9167f              verification     /m/0jbk         1.0
2833156  3fffcd2f80e7e639              verification   /m/01g317         1.0
2833158  3fffd718984ddbfb              verification   /m/01g317         1.0
2833159  3fffd718984ddbfb              verification   /m/01xyhv         1.0
2833161  3fffd718984ddbfb              verification    /m/09j2d         1.0
2833162  3fffd9a5ee5dbd62  crowdsource-verification    /m/026bk         1.0
2833163  3fffd9a5ee5dbd62              verification   /m/01g317         1.0
2833164  3fffd9a5ee5dbd62              verification   /m/01xyhv   

In [382]:
# next we count all the label names. Since we don't need the confidence column, we change it to count instead
trainable_label_counts = trainable_human_all.groupby('LabelName').count().rename(columns={'Confidence':'Counts'}).drop(columns=['ImageID', 'Source'])

In [383]:
trainable_label_counts = trainable_label_counts.reset_index()

In [385]:
# join the class descriptions
trainable_label_counts_sorted = trainable_label_counts.set_index('LabelName').join(class_descriptions.set_index('label_code')).sort_values('Counts', ascending=False)


In [386]:
trainable_label_counts_sorted.shape

(7172, 2)

7172 means that among 7178 trainable labels, we have 7172 trainable labels in the subset downloaded in my computer. Next, we select we labels whose counts are over 600. (of course we can change the number later)

In [388]:
trainable_label_counts_select = trainable_label_counts_sorted[trainable_label_counts_sorted['Counts'] > 600]

In [389]:
trainable_label_counts_select

Counts                       description
LabelName                                           
/m/01g317   807090                            Person
/m/09j2d    610840                          Clothing
/m/0dzct    331942                        Human face
/m/07j7r    315026                              Tree
/m/05s2s    266978                             Plant
/m/07yv9    228351                           Vehicle
/m/0cgh4    183125                          Building
/m/01prls   165525                      Land vehicle
/m/09j5n    134458                          Footwear
/m/0jbk     123529                            Animal
/m/0k4j     102685                               Car
/m/02wbm     84813                              Food
/m/0c9ph5    78846                            Flower
/m/083wq     77132                             Wheel
/m/0c_jw     68176                         Furniture
/m/03jm5     67167                             House
/m/0d4v4     59969                            Window
/m/03q69     56650                        Human hair
/m/04hgtk    49587                        Human head
/m/04bcr3    49232                             Table
/m/0463sg    48398                 Fashion accessory
/m/02p0tk3   45676                        Human body
/m/01xyhv    43949                              Suit
/m/01bqvp    42769                               Sky
/m/0jyfg     42028                           Glasses
/m/0dzf4     41589                         Human arm
/m/04rky     41522                            Mammal
/m/05y5lj    38801                  Sports equipment
/m/01lrl     35969                         Carnivore
/m/015p6     35586                              Bird
...            ...                               ...
/m/08tlbj      634                     Natural foods
/m/068hy       632                               Pet
/m/0f9_l       631                             Snail
/m/07pj7bq     630                 Bowling equipment
/m/03c7gz      629                       Door handle
/m/0n49352     629                       Sports gear
/m/0kpbc       625            Aerospace manufacturer
/m/02mf1n      624                            Gadget
/m/0898b       621                             Zebra
/m/01bqgn      621                     Bicycle frame
/m/0bpnmk0     620  Bicycles--Equipment and supplies
/m/012mj       619                Alcoholic beverage
/m/02vzx9      618                            Player
/m/03lq2       617                         Halloween
/m/0bxdx       615                          Windmill
/m/01n32       612                              City
/m/027n3_      611                           Eyebrow
/m/02f9f_      610                            Shower
/m/04jjh       610                            Letter
/m/068jd       609                        Photograph
/m/0ql23       609                        Pollinator
/m/0h8ly30     609           Automotive wheel system
/m/02068x      609                           Gondola
/m/02x984l     609                    Mechanical fan
/m/0bmgjqz     608                       Sport venue
/m/02tzth      607                      Post-it note
/m/06pj2k      605                               Lip
/m/05_5p_0     602               Table tennis racket
/m/0glmwd6     602              Communication Device
/m/0f9swq      601                              Chin

[447 rows x 2 columns]

So there are about 447 of the labels that appear more than 600 times in my subset. Next, we put the label name in the dataframe.

In [390]:
train_image_select = trainable_label_counts_select.join(trainable_human_all.set_index('LabelName'))

In [391]:
train_image_select

Counts        description           ImageID        Source  \
LabelName                                                              
/m/011_f4     648  String instrument  0000201cd362f303  verification   
/m/011_f4     648  String instrument  0000aa810854dc2e  verification   
/m/011_f4     648  String instrument  0000b3e5921ab7ff  verification   
/m/011_f4     648  String instrument  0000c33c6f4b8518  verification   
/m/011_f4     648  String instrument  0000d59fa570d973  verification   
/m/011_f4     648  String instrument  0002024f996741eb  verification   
/m/011_f4     648  String instrument  00027ae34cb09507  verification   
/m/011_f4     648  String instrument  0002a3c01c926a49  verification   
/m/011_f4     648  String instrument  0002eed6350f1840  verification   
/m/011_f4     648  String instrument  00036d5834d932bf  verification   
/m/011_f4     648  String instrument  0004359b4b703bcb  verification   
/m/011_f4     648  String instrument  0004c03cb6520680  verification   
/m/011_f4     648  String instrument  000548d21e085b90  verification   
/m/011_f4     648  String instrument  00058c84b5ef9dba  verification   
/m/011_f4     648  String instrument  0005e4b02ae05787  verification   
/m/011_f4     648  String instrument  0007e06651e9a48b  verification   
/m/011_f4     648  String instrument  0008abbbff45eb1e  verification   
/m/011_f4     648  String instrument  000918f221133f8f  verification   
/m/011_f4     648  String instrument  000952cdec3960fb  verification   
/m/011_f4     648  String instrument  0009fde62ded08a6  verification   
/m/011_f4     648  String instrument  000a1c0dcd207581  verification   
/m/011_f4     648  String instrument  000a817a350d8bdb  verification   
/m/011_f4     648  String instrument  000b65a36ad46f9e  verification   
/m/011_f4     648  String instrument  000b70a84aab664b  verification   
/m/011_f4     648  String instrument  000be19f13837eb0  verification   
/m/011_f4     648  String instrument  000cd8ac1345340b  verification   
/m/011_f4     648  String instrument  000da253e3b41fa8  verification   
/m/011_f4     648  String instrument  000f325c2e614758  verification   
/m/011_f4     648  String instrument  000ff2d34da4eb77  verification   
/m/011_f4     648  String instrument  0011179ae4c55b62  verification   
...           ...                ...               ...           ...   
/m/0zvk5    12396             Helmet  ff514e8d8bdc88ca  verification   
/m/0zvk5    12396             Helmet  ff57f2e88fb2e0e5  verification   
/m/0zvk5    12396             Helmet  ff58bcfab665e0a6  verification   
/m/0zvk5    12396             Helmet  ff5d0ce2e6ad2c31  verification   
/m/0zvk5    12396             Helmet  ff5de3d0c803f6e1  verification   
/m/0zvk5    12396             Helmet  ff64a99cdb51ac5e  verification   
/m/0zvk5    12396             Helmet  ff6cba46cb2d14ba  verification   
/m/0zvk5    12396             Helmet  ff6f9fcbb55a630d  verification   
/m/0zvk5    12396             Helmet  ff712df1d5875901  verification   
/m/0zvk5    12396             Helmet  ff7a542171e3251d  verification   
/m/0zvk5    12396             Helmet  ff7bfcad88d24180  verification   
/m/0zvk5    12396             Helmet  ff85f5012753da09  verification   
/m/0zvk5    12396             Helmet  ff953877863de084  verification   
/m/0zvk5    12396             Helmet  ff95efb364213f7d  verification   
/m/0zvk5    12396             Helmet  ff97d3c50bf253a9  verification   
/m/0zvk5    12396             Helmet  ff9cc21cf2ec2f17  verification   
/m/0zvk5    12396             Helmet  ffa3473f2a60cb20  verification   
/m/0zvk5    12396             Helmet  ffa66351c1f36eb9  verification   
/m/0zvk5    12396             Helmet  ffba42eee4135a11  verification   
/m/0zvk5    12396             Helmet  ffbb4e3ddf4de827  verification   
/m/0zvk5    12396             Helmet  ffc004c1b423c547  verification   
/m/0zvk5    12396             Helmet  ffcaa1565c11ba38  verification   
/m/0zvk5    12396             Helmet  ffd0350867a45

So there are 6304532 labels in total. We need to find how many images there are. (Again, since there might be more than 1 labels for an image) 

In [392]:
import os

In [393]:
datapath = list()
for root, dirs, files in os.walk('/media/yshi/4TB/openImage/train/'):
    for filename in files:
        nm, ext = os.path.splitext(filename)
        fullpath = os.path.join(os.path.abspath(root), filename)
        datapath.append((filename[:-4], fullpath))
datapath = pd.DataFrame(datapath, columns=['filename', 'fullpath'])
# print(df1)

In [394]:
datapath.shape

(493994, 2)

493994 are the total number of images in my subset. But how many of them can be used to train the 447 highly frequent labels above?

In [395]:
data_select = datapath.loc[datapath['filename'].isin(train_image_select['ImageID'])]

In [396]:
data_select

filename                                           fullpath
0       3cef2522b36f8c57  /media/yshi/4TB/openImage/train/3cef2522b36f8c...
1       3b4a15883505f0b6  /media/yshi/4TB/openImage/train/3b4a15883505f0...
2       1f7b221bfa9aacb9  /media/yshi/4TB/openImage/train/1f7b221bfa9aac...
3       0ae10d73844dea18  /media/yshi/4TB/openImage/train/0ae10d73844dea...
4       0b83dc18b027bc81  /media/yshi/4TB/openImage/train/0b83dc18b027bc...
5       3043cb9122c4d81c  /media/yshi/4TB/openImage/train/3043cb9122c4d8...
6       0cc625bbafb01cbe  /media/yshi/4TB/openImage/train/0cc625bbafb01c...
7       026f29e240a92aff  /media/yshi/4TB/openImage/train/026f29e240a92a...
8       2ca0b30070eb8e46  /media/yshi/4TB/openImage/train/2ca0b30070eb8e...
9       03cccddeded8eaa1  /media/yshi/4TB/openImage/train/03cccddeded8ea...
10      3e808a86e89916b7  /media/yshi/4TB/openImage/train/3e808a86e89916...
11      00484de7afe00a5d  /media/yshi/4TB/openImage/train/00484de7afe00a...
12      12c3e86b28589108  /media/yshi/4TB/openImage/train/12c3e86b285891...
13      34856e4405ad0689  /media/yshi/4TB/openImage/train/34856e4405ad06...
14      16028623d9ed3248  /media/yshi/4TB/openImage/train/16028623d9ed32...
15      25cb50508efc104f  /media/yshi/4TB/openImage/train/25cb50508efc10...
16      01323453fcd87855  /media/yshi/4TB/openImage/train/01323453fcd878...
17      0762f44048dc868e  /media/yshi/4TB/openImage/train/0762f44048dc86...
18      11e51b3ff4ee10ab  /media/yshi/4TB/openImage/train/11e51b3ff4ee10...
19      195eebe6716f828c  /media/yshi/4TB/openImage/train/195eebe6716f82...
20      3d25458877ff41e5  /media/yshi/4TB/openImage/train/3d25458877ff41...
21      0870e7ff70b42d8d  /media/yshi/4TB/openImage/train/0870e7ff70b42d...
22      3f322a0b2bdb72a6  /media/yshi/4TB/openImage/train/3f322a0b2bdb72...
24      322a0453bc24128c  /media/yshi/4TB/openImage/train/322a0453bc2412...
25      1a35da43974eaee6  /media/yshi/4TB/openImage/train/1a35da43974eae...
26      39fc761359241ecb  /media/yshi/4TB/openImage/train/39fc761359241e...
27      26f90ba3ffca3364  /media/yshi/4TB/openImage/train/26f90ba3ffca33...
28      2f80c2ae1031e267  /media/yshi/4TB/openImage/train/2f80c2ae1031e2...
29      01aa63506fe63226  /media/yshi/4TB/openImage/train/01aa63506fe632...
30      011f97f760fec57a  /media/yshi/4TB/openImage/train/011f97f760fec5...
...                  ...                                                ...
493963  25f1b047f5868fa8  /media/yshi/4TB/openImage/train/25f1b047f5868f...
493964  3f90ee259900253c  /media/yshi/4TB/openImage/train/3f90ee25990025...
493965  02a289b0659a1c01  /media/yshi/4TB/openImage/train/02a289b0659a1c...
493966  3372804f40284c71  /media/yshi/4TB/openImage/train/3372804f40284c...
493967  1e575ccf257fd5fd  /media/yshi/4TB/openImage/train/1e575ccf257fd5...
493968  100c05d343dae634  /media/yshi/4TB/openImage/train/100c05d343dae6...
493969  0bcbb259bb64d27f  /media/yshi/4TB/openImage/train/0bcbb259bb64d2...
493970  2e863c1508c2f3a9  /media/yshi/4TB/openImage/train/2e863c1508c2f3...
493971  174868b8e028de15  /media/yshi/4TB/openImage/train/174868b8e028de...
493972  12f65700890d55a1  /media/yshi/4TB/openImage/train/12f65700890d55...
493973  058a223db98d100a  /media/yshi/4TB/openImage/train/058a223db98d10...
493974  26a9da9244485d94  /media/yshi/4TB/openImage/train/26a9da9244485d...
493975  1a95d97f3ea0f72a  /media/yshi/4TB/openImage/train/1a95d97f3ea0f7...
493976  206def7448be3b74  /media/yshi/4TB/openImage/train/206def7448be3b...
493977  1071973a8d46722c  /media/yshi/4TB/openImage/train/1071973a8d4672...
493978  3705498b6606f7d2  /media/yshi/4TB/openImage/train/3705498b6606f7...
493979  08fd124be0e79a04  /media/yshi/4TB/openImage/train/08fd124be0e79a...
493980  267124944a00b55c  /media/yshi/4TB/openImage/train/267124944a00b5...
493981  000e83dc68db5ca8  /media/yshi/4TB/openImage/train/000e83dc68db5c...
493982  07e1a916d1a20090  /media/yshi/4TB/openImage/train/07e1a916d1a200...
493983  0ad9e5fcd74d49ec  /media/yshi/4TB/openImage/train/0ad9e5

There are 487973 rows 

In [397]:
train_image_select = train_image_select.loc[train_image_select['ImageID'].isin(datapath['filename'])]

In [398]:
train_image_select

Counts        description           ImageID        Source  \
LabelName                                                              
/m/011_f4     648  String instrument  0000201cd362f303  verification   
/m/011_f4     648  String instrument  0000aa810854dc2e  verification   
/m/011_f4     648  String instrument  0000b3e5921ab7ff  verification   
/m/011_f4     648  String instrument  0000c33c6f4b8518  verification   
/m/011_f4     648  String instrument  0000d59fa570d973  verification   
/m/011_f4     648  String instrument  0002024f996741eb  verification   
/m/011_f4     648  String instrument  00027ae34cb09507  verification   
/m/011_f4     648  String instrument  0002a3c01c926a49  verification   
/m/011_f4     648  String instrument  0002eed6350f1840  verification   
/m/011_f4     648  String instrument  00036d5834d932bf  verification   
/m/011_f4     648  String instrument  0004359b4b703bcb  verification   
/m/011_f4     648  String instrument  0004c03cb6520680  verification   
/m/011_f4     648  String instrument  000548d21e085b90  verification   
/m/011_f4     648  String instrument  00058c84b5ef9dba  verification   
/m/011_f4     648  String instrument  0005e4b02ae05787  verification   
/m/011_f4     648  String instrument  0007e06651e9a48b  verification   
/m/011_f4     648  String instrument  0008abbbff45eb1e  verification   
/m/011_f4     648  String instrument  000918f221133f8f  verification   
/m/011_f4     648  String instrument  000952cdec3960fb  verification   
/m/011_f4     648  String instrument  0009fde62ded08a6  verification   
/m/011_f4     648  String instrument  000a1c0dcd207581  verification   
/m/011_f4     648  String instrument  000a817a350d8bdb  verification   
/m/011_f4     648  String instrument  000b65a36ad46f9e  verification   
/m/011_f4     648  String instrument  000b70a84aab664b  verification   
/m/011_f4     648  String instrument  000be19f13837eb0  verification   
/m/011_f4     648  String instrument  000cd8ac1345340b  verification   
/m/011_f4     648  String instrument  000da253e3b41fa8  verification   
/m/011_f4     648  String instrument  000f325c2e614758  verification   
/m/011_f4     648  String instrument  000ff2d34da4eb77  verification   
/m/011_f4     648  String instrument  0011179ae4c55b62  verification   
...           ...                ...               ...           ...   
/m/0zvk5    12396             Helmet  3f4ee313c600abc5  verification   
/m/0zvk5    12396             Helmet  3f50c3040000771d  verification   
/m/0zvk5    12396             Helmet  3f54903524b8ee14  verification   
/m/0zvk5    12396             Helmet  3f556be734c30f7c  verification   
/m/0zvk5    12396             Helmet  3f5b542ba80b2e97  verification   
/m/0zvk5    12396             Helmet  3f675dae183dc253  verification   
/m/0zvk5    12396             Helmet  3f6cd801beeeb263  verification   
/m/0zvk5    12396             Helmet  3f73f5a0dabcfce6  verification   
/m/0zvk5    12396             Helmet  3f78d91e4f3af1c5  verification   
/m/0zvk5    12396             Helmet  3f7ff29adb6db99e  verification   
/m/0zvk5    12396             Helmet  3f85c3c772995cee  verification   
/m/0zvk5    12396             Helmet  3f8d6fbd025753e2  verification   
/m/0zvk5    12396             Helmet  3f94a7278e462bf6  verification   
/m/0zvk5    12396             Helmet  3f9765763ae60feb  verification   
/m/0zvk5    12396             Helmet  3f9766b90e1f476c  verification   
/m/0zvk5    12396             Helmet  3f98a62b95ad6348  verification   
/m/0zvk5    12396             Helmet  3f9ad3cd15e70f53  verification   
/m/0zvk5    12396             Helmet  3f9d6b10dcca069e  verification   
/m/0zvk5    12396             Helmet  3f9fdac074391743  verification   
/m/0zvk5    12396             Helmet  3fa432d4d2c48244  verification   
/m/0zvk5    12396             Helmet  3fa96ecf6d752469  verification   
/m/0zvk5    12396             Helmet  3faaf9985383bd0e  verification   
/m/0zvk5    12396             Helmet  3fb378940ba28

So in these 487973 images there are 1868410 labels. Average 4 each.